### PR TITLE
Safer AmberTools version check

### PR DESCRIPTION
**Pull request by mattwthompson**
#1865

- [x] Tag issue being addressed
- [ ] Add [tests](https://github.com/openforcefield/openff-toolkit/tree/main/openff/toolkit/_tests)
- [ ] Update docstrings/[documentation](https://github.com/openforcefield/openff-toolkit/tree/main/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [ ] Update [changelog](https://github.com/openforcefield/openff-toolkit/blob/main/docs/releasehistory.rst)

### CODE REVIEW

File: openff/toolkit/utils/ambertools_wrapper.py [hunk 1]
```diff
@@ -69,10 +70,7 @@ def __init__(self):
                 f"available. {self._toolkit_installation_instructions}"
             )
 
-        # TODO: More reliable way to extract AmberTools version
-        out = subprocess.check_output(["antechamber", "-L"])
-        ambertools_version = out.decode("utf-8").split("\n")[1].split()[3].strip(":")
-        self._toolkit_version = ambertools_version
+        self._toolkit_version = get_ambertools_version()
```
Comment (j-wags) [inline openff/toolkit/utils/ambertools_wrapper.py:73]: For reference, source code is here: https://github.com/openforcefield/openff-utilities/blob/ee9f969c252d02548c08537fb34cb713923eef70/openff/utilities/provenance.py#L39-L42

**Review by j-wags — APPROVED** [review]
Thanks @mattwthompson - I'll update releasenotes and merge.

**Comment by pbuslaev** [comment]
Maybe a bit late, but the current solution does not work well in combination with slurm. The use case is as following:

I want to assign charges with `ambertools` to a molecule and I am submitting a script that is doing that to the external cluster with [sbatch](https://slurm.schedmd.com/sbatch.html). By default (check --export documentation for sbatch) local environment variables (including PATH) will also be exported to cluster. And the problem occurs, when conda path is different at local and server node (this can be quite a frequent scenario when running on the cloud). In this case, `get_ambertools_version`, which is [looking for `conda` path](https://github.com/openforcefield/openff-utilities/blob/90d49544c6f2c131d7c3edbaef68ade9883d7861/openff/utilities/provenance.py#L39) will fail. While there are ways around it on the user side, it took me quite a while to detect the issue. I wonder if there is a way on the openff side to solve it? For instance, [`_get_conda_list_package_versions()`](https://github.com/openforcefield/openff-utilities/blob/90d49544c6f2c131d7c3edbaef68ade9883d7861/openff/utilities/provenance.py#L7) can in addition check the output of `conda info --base` or smth like this?

**Comment by mattwthompson** [comment]
> conda path is different at local and server node

Doesn't this by definition mean different environments are being used?

I'm open to stuffing something through to `_get_conda_list_package_versions` but this is the sort of thing that can have side effects

**Pull request merged**